pygitguardian/client.py

```diff
@@ -514,3 +516,29 @@ def read_metadata(self) -> Optional[Detail]:
 
         self.secret_scan_preferences = metadata.secret_scan_preferences
         return None
+
+    def create_jwt(
+        self,
+        jwt_audience: str,
+        service: JWTService,
+        extra_headers: Optional[Dict[str, str]] = None,
+    ) -> Union[Detail, JWTResponse]:
+        """
+        Create a JWT token for other GitGuardian services.
+
+        Only used to interact with HasMySecretLeaked for now.
+        :return: Detail or JWT response and status code
+        """
+
+        resp = self.post(
+            endpoint="auth/jwt",
+            data={"audience": jwt_audience, "audience_type": service.value},
+            extra_headers=extra_headers,
+        )
+        obj: Union[Detail, JWTResponse]
+        if is_ok(resp):
+            obj = cast(JWTResponse, JWTResponse.SCHEMA.load(resp.json()))
```

Comment:
I just merged some changes in py-gitguardian which you can take advantage of here: if you make `JWTResponse` inherit from `FromDictMixin` in addition to `Base`, then this code can be rewritten as:

```
obj = JWTResponse.from_dict(resp.json())
```

This is simpler and more type safe.